app/controllers/ProductController.scala

```diff
@@ -14,21 +25,46 @@ import javax.inject.Inject
  */
 class ProductController @Inject() (
                                     cc: ControllerComponents,
-                                    environment: Environment
+                                    environment: Environment,
                                   ) extends AbstractController(cc) {
-  // Create the format for the product in JSON form.
-  implicit val productFormat = Json.format[Product]
+
+
+  val xa = Transactor.fromDriverManager[IO](
+    "org.postgresql.Driver", // driver classname
+    "jdbc:postgresql:bakery_db", // connect URL (driver-specific)
+    "postgres", // user
+    "" // password
+  )
 
   // When the user route to <root>/pass-bakery/status then this will return a status JSON
-  def getStatus = Action {
+  def getStatus: Action[AnyContent] = Action {
 
     // Used LocalDateTime since it time and date is formatted in ISO-8601 as it is written in its doc.
     val curDateTime = LocalDateTime.now()
 
     val curEnv = environment.mode
-    val product = Product("pass-bakery", curEnv.toString, curDateTime.toString)
+    val status = EndPointStatus("pass-bakery", curEnv.toString, curDateTime.toString)
 
     // Send OK signal with JSON status
-    Ok(Json.toJson(product))
+    Ok(Json.toJson(status))
+  }
+
+  // When the user route to <root>/rest/bakery/products then this will return all the elements of the bakery
+  // database.
+  def getInventory: Action[AnyContent] = Action {
+
+    // Select * statement with rows returned as a list so I can put it in JSON format
+    def getResults[T: Read]: List[T] = {
+        sql"select * from Bakery".query[T] // Query0[*]
+          .to[List] // ConnectionIO[List[Product]]
+          .transact(xa) // IO[List[Product]]
+          .unsafeRunSync() // List[Product]
+          .take(5) // List[Product]
+    }
+
+    val results : List[Product] = getResults[Product]
+
+
+    Ok(Json.toJson(results.toString()))
```

Comment:
Notice how you are mixing responsibilities in the file. you have both query logic and result serialization to an HTTP response. In real code we separate this and have our controller ask for a response, not handle the business logic itself, generally with a helper class. Declare a class called ProductDAO that take responsibility to instantiate a connection with the database and exposes a `getResults` function that houses the query logic (this means the transactor, xa variable above, will be instantiated there as well.

Reply:
That makes sense, I've made a new services folder and added ProductDAO with the query logic in it!